src/axom/inlet/YAMLReader.cpp

```diff
@@ -0,0 +1,382 @@
+{
+  return getArray(id, values);
+}
+
+bool YAMLReader::getDoubleMap(const std::string& id,
+                              std::unordered_map<int, double>& values)
+{
+  return getArray(id, values);
+}
+
+bool YAMLReader::getBoolMap(const std::string& id,
+                            std::unordered_map<int, bool>& values)
+{
+  return getArray(id, values);
+}
+
+bool YAMLReader::getStringMap(const std::string& id,
+                              std::unordered_map<int, std::string>& values)
+{
+  return getArray(id, values);
+}
+
+bool YAMLReader::getIntMap(const std::string& id,
+                           std::unordered_map<std::string, int>& values)
+{
+  return getDictionary(id, values);
+}
+
+bool YAMLReader::getDoubleMap(const std::string& id,
+                              std::unordered_map<std::string, double>& values)
+{
+  return getDictionary(id, values);
+}
+
+bool YAMLReader::getBoolMap(const std::string& id,
+                            std::unordered_map<std::string, bool>& values)
+{
+  return getDictionary(id, values);
+}
+
+bool YAMLReader::getStringMap(const std::string& id,
+                              std::unordered_map<std::string, std::string>& values)
+{
+  return getDictionary(id, values);
+}
+
+bool YAMLReader::getIndices(const std::string& id, std::vector<int>& indices)
+{
+  indices.clear();
+  const auto node = detail::traverseNode(m_root, id);
+  if(!node.dtype().is_list())
+  {
+    return false;
+  }
+  indices.resize(node.number_of_children());
+  // Arrays in YAML are contiguous so we don't need to query the input file
+  std::iota(indices.begin(), indices.end(), 0);
+  return true;
+}
+
+bool YAMLReader::getIndices(const std::string& id,
+                            std::vector<std::string>& indices)
+{
+  indices.clear();
+  const auto node = detail::traverseNode(m_root, id);
+  if(!node.dtype().is_object())
+  {
+    return false;
+  }
+  // FIXME: Update to range-based for loops when Axom begins using Conduit 0.6.0
+  auto itr = node.children();
+  while(itr.has_next())
+  {
+    const auto& child = itr.next();
+    indices.push_back(child.name());
+  }
+  return true;
+}
+
+FunctionVariant YAMLReader::getFunction(const std::string&,
+                                        const FunctionType,
+                                        const std::vector<FunctionType>&)
+{
+  SLIC_ERROR("[Inlet] YAML does not support functions");
+  return {};
+}
+
+template <typename T>
+bool YAMLReader::getDictionary(const std::string& id,
+                               std::unordered_map<std::string, T>& values)
+{
+  values.clear();
+  const auto node = detail::traverseNode(m_root, id);
+  if(!node.dtype().is_object())
+  {
+    return false;
+  }
+
+  // FIXME: Update to range-based for loops when Axom begins using Conduit 0.6.0
+  auto itr = node.children();
+  while(itr.has_next())
+  {
+    const auto& child = itr.next();
+    const auto name = child.name();
+    if(!getValue(child, values[name]))
```

Comment:
I had to think about this for a minute to understand what was going on. For me, it would have been simpler like this:
```
T value;
if (getValue(child, value)) {
    values[name] = value;
}
```

Reply:
Thanks for the suggestion - I agree that this a lot more intuitive.